packages/core/src/utilities/validation.js

```diff
@@ -344,18 +344,17 @@ export const validateField: ValidateField = (
   let errorMessages = [];
   if (visible) {
     const value = getValueFromField(field);
+    const valueProvided = hasValue(value);
     if (required) {
-      isValid = hasValue(value);
-      if (!isValid) {
+      if (!valueProvided) {
+        isValid = valueProvided;
         const { missingValueMessage = "A value must be provided" } = field;
         errorMessages.push(missingValueMessage);
       }
-    }
-
-    if (!hasValue(value)) {
+    } else if (!valueProvided) {
       // do not run all validations if the field is empty
       return Object.assign({}, field, {
-        isValid,
+        isValid: true,
```

Comment:
This is the key change - ensuring that validity is true in this condition, however it needed to be moved to an else block of the basic requirement check. Optional fields that have validation rules will only be validated if a value is provided - I think this is the intention of the previous change, @vasinkevych please correct me if I'm wrong here.

Reply:
looks good, I believe it should work

Reply:
thanks